infra/provider.tf

```diff
@@ -19,7 +19,7 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 4.84"
+      version = "= 4.28.0"
```

Comment:
I'm using hashicorp/google Provider version 4.28.0 because that's where the fix initially landed.

Reply:
Even with hashicorp/google Provider version 4.28.0, we get the "network ... # forces replacement" error.
Relevant [logs here](https://pantheon.corp.google.com/cloud-build/builds/828aaf0d-f337-4da2-b11d-c4efa0c5dd36;step=3?e=13803378&mods=notebooks_use_staging_api&project=cloud-foundation-cicd):
```
- Installing hashicorp/google v4.28.0...
```
```
~ network           = "xwiki-gce" -> "projects/ ... /global/networks/projects/ ... /global/networks/xwiki-gce" # forces replacement
```